core/org.osate.xtext.aadl2/src/org/osate/xtext/aadl2/validation/Aadl2JavaValidator.java

```diff
@@ -3145,7 +3145,7 @@ private void checkComponentTypeModes(ComponentType componentType) {
 		} else if (componentType.isDerivedModes()) {
 			INode requiresModesNode = getRequiresModesNode(componentType);
 			if (componentType.getExtended() != null && componentType.getExtended().getAllModes().stream()
-					.anyMatch(extendedMode -> !extendedMode.isDerived())) {
+					.anyMatch(extendedMode -> !extendedMode.isDerived()) && requiresModesNode != null) {
```

Comment:
It is a bit unfortunate that SonarCloud is complaining about a potential NPE for `requiresModesNode`. It is guaranteed to not be null because `componentType.isDerivedModes()` is true, but SonarCloud has no way of knowing that. I don't like the idea of including a null check in the conditional after `anyMatch` because I think it confuses the logic of the validator. When reading validation code and trying to understand what it is doing, the if-statements are critical to figuring out what the potential cases are that the validation method can deal with. I don't like the idea of mixing legitimate conditionals with a conditional that will always evaluate to true and is only there because of SonarCloud.

I would suggest moving the declaration of `requiresModesNode` down into the if body and then doing a separate null check before the call to `acceptError`. Also, put in a comment stating that `requiresModesNode` should never be null and that this check is only there to satisfy SonarCloud.